jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -46,9 +46,28 @@ typedef struct
 
 /**
  * Round high part of 128-bit integer to uint64_t
+ *
+ * @return rounded high to uint64_t
  */
-#define ECMA_UINT128_ROUND_HIGH_TO_UINT64(name) \
-  (name.hi + (name.lo >> 63u))
+static uint64_t
+ecma_round_high_to_uint64 (ecma_uint128_t *num_p)
+{
+  uint64_t masked_lo = num_p->lo & ~(1ULL << 63u);
+  uint64_t masked_hi = num_p->hi & 0x1;
+
+  if (num_p->lo >> 63u != 0)
```

Comment:
Seems like all of these if's could be merged together into a single condition
```
if ((num_p->lo >> 63u != 0)
     && (masked_lo > 0 || masked_hi != 0))
```
What do you think?

Reply:
I totally overlooked that I could merge those. That's a neat idea indeed. :)